sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SerializationDiagnosticsContext.java

```diff
@@ -0,0 +1,90 @@
+        }
+
+        serializationDiagnosticsList.add(serializationDiagnostics);
+    }
+
+    public <T> T getResource(Callable<T> function, SerializationType serializationType) {
+        ZonedDateTime serializationStartTime = ZonedDateTime.now(ZoneOffset.UTC);
+        T t = null;
+        try {
+            t = function.call();
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+
+        ZonedDateTime serializationEndTime = ZonedDateTime.now(ZoneOffset.UTC);
+        if (serializationDiagnosticsList == null) {
+            serializationDiagnosticsList = new ArrayList<>();
+        }
+
+        serializationDiagnosticsList.add(new SerializationDiagnostics(serializationStartTime, serializationEndTime, serializationType));
+        return t;
+    }
+
+    @JsonSerialize(using = SerializationDiagnosticsContext.SerializationDiagnosticsSerializer.class)
+    public static class SerializationDiagnostics {
+        public volatile ZonedDateTime startTimeUTC;
+        public volatile ZonedDateTime endTimeUTC;
+        public volatile SerializationDiagnosticsContext.SerializationType serializationType;
+
+        public SerializationDiagnostics(ZonedDateTime startTimeUTC, ZonedDateTime endTimeUTC, SerializationDiagnosticsContext.SerializationType serializationType) {
+            this.startTimeUTC = startTimeUTC;
+            this.endTimeUTC = endTimeUTC;
+            this.serializationType = serializationType;
+        }
+    }
+
+    static class SerializationDiagnosticsSerializer extends StdSerializer<SerializationDiagnosticsContext.SerializationDiagnostics> {
+
+        public SerializationDiagnosticsSerializer() {
+            super(SerializationDiagnosticsContext.SerializationDiagnostics.class);
+        }
+
+        @Override
+        public void serialize(SerializationDiagnosticsContext.SerializationDiagnostics serializationDiagnostics, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
+            Duration durationinMS = serializationDiagnostics.startTimeUTC == null ?
+                null : serializationDiagnostics.endTimeUTC == null ?
+                Duration.ZERO : Duration.between(serializationDiagnostics.startTimeUTC, serializationDiagnostics.endTimeUTC);
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeObjectField("serializationType", serializationDiagnostics.serializationType);
+            jsonGenerator.writeStringField("startTimeUTC", ZonedDateTimeSerializer.formatDateTime(serializationDiagnostics.startTimeUTC));
+            jsonGenerator.writeStringField("endTimeUTC", ZonedDateTimeSerializer.formatDateTime(serializationDiagnostics.endTimeUTC));
+            if(durationinMS != null) {
+                jsonGenerator.writeNumberField("durationInMicroSec", durationinMS.toNanos()/1000);
+            }
+
+            jsonGenerator.writeEndObject();
+        }
+    }
+    public enum  SerializationType{
```

Comment:
code style: always a space between a term and "{"
also you have two spaces between enum and SerializationType please fix

`public enum  SerializationType{ `-> `public enum SerializationType {`